code/entities/machine/soda-fountain/SodaFountain.CupFillable.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class CupFillable : ModelEntity
+{
+    public enum MaterialGroup : int
+    {
+        Default = 0,
+        Blue = 1,
+        Green = 2,
+        Red = 3,
+        Black = 4
+    }
+
+    static public string CupItemUniqueId = "soda"; 
+    static public string CupItemUniqueIdConk = "soda-conk"; 
+    static public string CupItemUniqueIdMionPisz = "soda-mionpisz"; 
+    static public string CupItemUniqueIdSpooge = "soda-spooge"; 
+
+    public Dispenser DispenserParent { get; set; }
+    private bool IsAssembled { get; set; }
+
+    public CupFillable() //For the compiler...
+    {
+       
+    }
+    public CupFillable(Dispenser dispenserParent)
+    {
+        DispenserParent = dispenserParent;
+        DispenserParent.Cup = this;
+    }
+    public override void Spawn()
+    {
+        base.Spawn();
+
+        Transmit = TransmitType.Always;
+
+        SetModel("models/papercup/walker/sodafountaincup/papercup.vmdl");
+
+        SetupPhysicsFromModel(PhysicsMotionType.Static);
+    }
+    public void SetCupColor(SodaFountain.SodaType type)
```

Comment:
When refactoring this class to store the `SodaType` instead of the `CupType`, this shoudl be renamed `SetSodaType`

Reply:
Moved this into the Initialize function, where cup's material group gets set based on the soda type